params/astria_config.go

```diff
@@ -0,0 +1,370 @@
+package params
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"math/big"
+	"sort"
+
+	"github.com/btcsuite/btcd/btcutil/bech32"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/rlp"
+)
+
+type AstriaForks struct {
+	orderedForks []AstriaForkData // sorted in descending order by Height
+	forkMap      map[string]AstriaForkConfig
+}
+
+type AstriaForkConfig struct {
+	Height            uint64                      `json:"height"`
+	Halt              bool                        `json:"halt,omitempty"`
+	SnapshotChecksum  string                      `json:"snapshotChecksum,omitempty"`
+	ExtraDataOverride hexutil.Bytes               `json:"extraDataOverride,omitempty"`
+	FeeCollector      *common.Address             `json:"feeCollector,omitempty"`
+	EIP1559Params     *AstriaEIP1559Params        `json:"eip1559Params,omitempty"`
+	Sequencer         *AstriaSequencerConfig      `json:"sequencer,omitempty"`
+	Celestia          *AstriaCelestiaConfig       `json:"celestia,omitempty"`
+	BridgeAddresses   []AstriaBridgeAddressConfig `json:"bridgeAddresses,omitempty"`
+}
+
+type AstriaForkData struct {
+	Name                string
+	Height              uint64
+	Halt                bool
+	SnapshotChecksum    string
+	ExtraDataOverride   hexutil.Bytes
+	FeeCollector        common.Address
+	EIP1559Params       AstriaEIP1559Params
+	Sequencer           AstriaSequencerConfig
+	Celestia            AstriaCelestiaConfig
+	BridgeAddresses     map[string]*AstriaBridgeAddressConfig // astria bridge addess to config for that bridge account
+	BridgeAllowedAssets map[string]struct{}                   // a set of allowed asset IDs structs are left empty
+}
+
+type AstriaSequencerConfig struct {
+	ChainID           string `json:"chainId"`
+	AddressPrefix     string `json:"addressPrefix,omitempty"`
+	StartHeight       uint32 `json:"startHeight"`
+	StopHeight        uint32 `json:"-"`
+	RollupStartHeight uint64 `json:"-"`
+}
+
+type AstriaCelestiaConfig struct {
+	ChainID        string `json:"chainId"`
+	StartHeight    uint64 `json:"startHeight"`
+	HeightVariance uint64 `json:"heightVariance"`
+}
+
+type AstriaEIP1559Params struct {
+	MinBaseFee               uint64 `json:"minBaseFee"`
+	ElasticityMultiplier     uint64 `json:"elasticityMultiplier"`
+	BaseFeeChangeDenominator uint64 `json:"baseFeeChangeDenominator"`
+}
+
+func (c *ChainConfig) AstriaExtraData(height uint64) []byte {
+	fork := c.GetAstriaForks().GetForkAtHeight(height)
+	if fork.ExtraDataOverride != nil {
+		return fork.ExtraDataOverride
+	}
+
+	// create default extradata
+	extra, _ := rlp.EncodeToBytes([]interface{}{
+		c.AstriaRollupName,
+		fork.Sequencer.StartHeight,
+		fork.Celestia.StartHeight,
+		fork.Celestia.HeightVariance,
+	})
```

Comment:
I think we should use basically the whole Fork here. We could RLP encode it so it could be read, or just hash it?

Reply:
updated in commit #[14592f0](https://github.com/astriaorg/astria-geth/pull/59/commits/14592f08ce381f62964184c562c39d2ffbd01e79) to be a keccak-256 hash of the fork config if not overridden